From tklauser@sym.noone.org Wed Dec 20 02:42:41 2006
Date: Wed, 20 Dec 2006 11:42:12 +0100
From: Tobias Klauser <tklauser@distanz.ch>
To: kernel-janitors@lists.osdl.org
Cc: gregkh@suse.de, linux-usb-devel@lists.sourceforge.net
Subject: USB: Remove unneeded void * casts in idmouse.c
Message-ID: <20061220104212.GA14334@distanz.ch>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

The patch removes unneeded void * casts for the following (void *) pointers:
- struct file: private_data

The patch also contains some whitespace and coding style cleanups in the
relevant areas.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/idmouse.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- gregkh-2.6.orig/drivers/usb/misc/idmouse.c
+++ gregkh-2.6/drivers/usb/misc/idmouse.c
@@ -269,7 +269,7 @@ static int idmouse_release(struct inode 
 	/* prevent a race condition with open() */
 	mutex_lock(&disconnect_mutex);
 
-	dev = (struct usb_idmouse *) file->private_data;
+	dev = file->private_data;
 
 	if (dev == NULL) {
 		mutex_unlock(&disconnect_mutex);
@@ -304,17 +304,15 @@ static int idmouse_release(struct inode 
 static ssize_t idmouse_read(struct file *file, char __user *buffer, size_t count,
 				loff_t * ppos)
 {
-	struct usb_idmouse *dev;
+	struct usb_idmouse *dev = file->private_data;
 	int result;
 
-	dev = (struct usb_idmouse *) file->private_data;
-
 	/* lock this object */
-	down (&dev->sem);
+	down(&dev->sem);
 
 	/* verify that the device wasn't unplugged */
 	if (!dev->present) {
-		up (&dev->sem);
+		up(&dev->sem);
 		return -ENODEV;
 	}
 
